sidecar/sokuji_sidecar/native_models.py

```
@@ -331,28 +341,81 @@ async def download(model_id, send, should_cancel=None, repo=None):
                          if llama_runtime.binary_path(f) is None]
-        total += len(llama_flavors)
-    done = 0
-    for r, fname in files:
+        total_units += len(llama_flavors)
+
+    # Byte mode when the total size is known (all catalog cards). Flavor
+    # installs weight in at a nominal estimate; the final event pins to total.
+    size = None
+    try:
+        size = model_size(model_id if not repo else repo)
+    except Exception:
```

Ruff (0.15.20)

[warning] Do not catch blind exception: `Exception`

(BLE001)